frontend/src/lib/components/Sidebar/Sidebar.svelte

```diff
@@ -0,0 +1,323 @@
+<!-- Navigation Toggle -->
+<button
+    type="button"
+    class="align-middle text-gray-500 hover:text-gray-600"
+    data-hs-overlay="#docs-sidebar"
+    aria-controls="docs-sidebar"
+    aria-label="Toggle navigation"
+>
+    <span class="sr-only">Navigate Chapters</span>
+    <svg
+        xmlns="http://www.w3.org/2000/svg"
+        fill="none"
+        viewBox="0 0 24 24"
+        stroke-width="1.5"
+        stroke="currentColor"
+        class="h-6 w-6"
+    >
+        <path
+            stroke-linecap="round"
+            stroke-linejoin="round"
+            d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
+        />
+    </svg>
+</button>
+<!-- End Navigation Toggle -->
+
+<div
+    id="docs-sidebar"
+    class="hs-overlay hs-overlay-open:translate-x-0 fixed bottom-0 start-0 top-0 z-[60] hidden w-64 -translate-x-full transform overflow-y-auto border-e border-gray-200 bg-white pb-10 pt-7 transition-all duration-300 dark:border-gray-700 dark:bg-gray-800 [&::-webkit-scrollbar-thumb]:rounded-full [&::-webkit-scrollbar-thumb]:bg-gray-300 dark:[&::-webkit-scrollbar-thumb]:bg-slate-500 [&::-webkit-scrollbar-track]:bg-gray-100 dark:[&::-webkit-scrollbar-track]:bg-slate-700 [&::-webkit-scrollbar]:w-2"
+>
+    <div class="px-6">
+        <a class="flex-none text-xl font-semibold dark:text-white" href="/" aria-label="Chapters"
+            >Chapters</a
+        >
+    </div>
+    <nav
+        class="hs-accordion-group flex w-full flex-col flex-wrap p-6 overflow-x-visible"
+        data-hs-accordion-always-open
+    >
+        <ul class="space-y-1.5">
+            <li>
+                <a
+                    class="flex items-center gap-x-3.5 rounded-lg bg-gray-100 px-2.5 py-2 text-sm text-slate-700 hover:bg-gray-100 dark:bg-gray-900 dark:text-white dark:hover:bg-gray-900 dark:focus:outline-none dark:focus:ring-1 dark:focus:ring-gray-600"
```

Comment:
for each individual <li></li>, we can make a separate component for reusable purpose in the future.